cornell-dti/cu-apts
# Apartment cards fix: number of reviews, routes

### Summary 






This PR closes #125 and makes the following changes for apartment cards: 
- Changes route `page-data/:page` to retrieve the number of reviews for the apartment, not the landlord associated with the apartment
- Cards say "1 Review" instead of "1 Reviews"
- Clicking on a card or a search result routes to the apartment page, i.e. `apartment/:aptId` instead of the landlord page, i.e. `landlord/:landlordId`
- Some minor code clean up: destructuring in `Autocomplete.tsx`

Note: Many cards say "0 Reviews" because these apartments don't have any reviews, and the apartments that have reviews are not shown because they don't happen to be one of the first 3 (home page) or first 12 (reviews page) buildings in the `buildings` collection. We can address this in another PR, i.e. show most reviewed apartments. 

https://user-images.githubusercontent.com/57203639/144352479-d2bf78dd-f8c0-40a2-8ada-861383d17b04.mov


### Test Plan 
- Click on an apartment card from the `/reviews` page and home`/` page. Test that it goes to `apartment/:aptId`. 
- Click on a search result. Test that it goes to `apartment/:aptId`. 
- Check that addresses do not appear on a search result if the name of the building is the same as the address, e.g. 312 College Ave.
